src/org/sage_one_sample/sageone/SageoneSigner.java

```diff
@@ -0,0 +1,130 @@
+package org.sage_one_sample.sageone;
+
+import java.net.*;
+import java.util.*;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+public class SageoneSigner {
+  String method;
+  String url;
+  TreeMap<String, String> params;
+  String nonce;
+  String secret;
+  String token;
+
+  SageoneSigner(String requestMethod, String requestUrl, TreeMap<String, String> requestBodyParams, String requestNonce, String signingSecret, String accessToken) {
+    method = requestMethod;
+    url = requestUrl;
+    params = requestBodyParams;
+    nonce = requestNonce;
+    secret = signingSecret;
+    token = accessToken;
+  }
+
+  public String signature() {
+    String result = null;
+    try {
+      result = Base64.getEncoder().encodeToString(hmacSha1());
+    } catch (UnsupportedEncodingException | NoSuchAlgorithmException | InvalidKeyException e) {
+      //handle exceptions
```

Comment:
Print to console with the error perhaps? So the user knows something has gone wrong.